### ISSUE TITLE
Column cannot be resolved for views

### ISSUE THREAD

**Issue by sergiogabriel** [prestodb/presto#9173]
Steps to reproduce
```
create table memory.default.t (s varchar);
create view memory.default.v as select s from memory.default.t;

select memory.default.t.s from memory.default.t; -- works
select memory.default.v.s from memory.default.v; -- fails with: Column 'memory.default.v.s' cannot be resolved
```

-----------------

I was integrating presto (0.186) with metabase (0.26.1) and in my environment I have created a catalog just to create views that maps my business rules. So when metabase tries to submit any query that projects columns in format "\<schema\>"."\<view\>"."\<column\>", presto returns the following error: **Column '\<schema\>.\<view\>.\<column\>' cannot be resolved**.

I have tried several times in several different ways to discover the origin of that error. 
But the conclusion is it just occurs with views.

After debugging presto project I found a possible solution.

Follow the code:

In presto-main/src/main/java/com/facebook/presto/sql/analyzer/Field.java:

```
    public boolean matchesPrefix(Optional<QualifiedName> prefix)
    {
        return !prefix.isPresent() || relationAlias.isPresent() && relationAlias.get().hasSuffix(prefix.get());
    }

```

was changed to:

```
    public boolean matchesPrefix(Optional<QualifiedName> prefix)
    {
        return !prefix.isPresent() || relationAlias.isPresent() && relationAlias.get().hasSuffix(prefix.get()) || ((Supplier<Boolean>) () -> {
                if (!originTable.isPresent())
                    return false;

                int idx = 0;
                boolean match = true;

                QualifiedObjectName origin = originTable.get();
                List<String> prefixes = prefix.get().getParts();

                switch (prefixes.size()) {
                    case 3: match =          origin.getCatalogName().equalsIgnoreCase(prefixes.get(idx++));
                    case 2: match = match && origin.getSchemaName().equalsIgnoreCase(prefixes.get(idx++));
                    case 1: match = match && origin.getObjectName().equalsIgnoreCase(prefixes.get(idx++));
                }

                return idx > 0 && match;
            }).get();
    }

```

In my tests it is working well.

-- SG

**Comment by roborohit** [comment]
+1

I'm seeing same issue when connecting Metabase (v 0.26.1) to Presto (0.148)

**Comment by satishvarmadandu** [comment]
How can we update this jdbc and integrate with Metabase to solve this issue. Questions part is not working for us due to this

**Comment by findepi** [comment]
@sergiogabriel i think you cannot reference column using `schema_name.view_name.column_name` syntax.. But I might be as well misunderstanding the problem at hand.
Could you provide an example query reproducing the problem? (You can find many good examples on how to do this, e.g. https://github.com/prestodb/presto/issues/9215)

**Comment by ricardoekm** [comment]
I'm facing the same problem here. The query on the original table works, however on the view it doesn't:

-- View
CREATE OR REPLACE VIEW linha_aprovada AS
SELECT data
FROM pedido.linha

-- Query on the original table, it works
SELECT pedido.linha.data FROM pedido.linha 

-- Query on the view, it fails
SELECT pedido.linha_aprovada.data FROM pedido.linha_aprovada 

"Column 'pedido.linha_aprovada.data' cannot be resolved"

**Comment by avnersentinelone** [comment]
The suggested change above worked for me too, it will be great if it could be inserted 
it fixed my metabase issue

**Comment by MichaelZhao9824** [comment]
I got the same issue with metabase as well. And it has been fixed using the code above. Not sure if there is a plan to do MR for the change

**Comment by rschlussel** [comment]
Fixed by #12565
